venafi/resource_venafi_ssh_certificate_test.go

```diff
@@ -60,7 +96,87 @@ func TestSshCert(t *testing.T) {
 	data := getTestData()
 	data.publicKeyMethod = "service"
 
-	config := fmt.Sprintf(tppSshCertResourceTest, tokenSshCertProv, data.keyId, data.template, data.publicKeyMethod, data.validityPeriod, data.sourceAddress)
+	// data.principals only holds the values for principals, actually we are testing "principal" attribute defined at the resource.
+	config := fmt.Sprintf(tppSshCertResourceTest, tokenSshCertProv, data.keyId, data.template, data.publicKeyMethod, data.validityPeriod, data.principals, data.sourceAddress)
+	t.Logf("Testing SSH certificate with config:\n %s", config)
+	r.Test(t, r.TestCase{
+		Providers: testProviders,
+		Steps: []r.TestStep{
+			r.TestStep{
+				Config: config,
+				Check: func(s *terraform.State) error {
+					err := checkSshCertificate(t, &data, s)
+					if err != nil {
+						return err
+					}
+					return nil
+				},
+				ExpectNonEmptyPlan: true,
+			},
+		},
+	})
+}
+
+func TestSshCertNewAttrPrincipals(t *testing.T) {
+	t.Log("Testing creating ssh certificate with new attribute for principals")
+
+	data := getTestData()
+	data.publicKeyMethod = "service"
+
+	config := fmt.Sprintf(tppSshCertResourceTestNewAttrPrincipals, tokenSshCertProv, data.keyId, data.template, data.publicKeyMethod, data.validityPeriod, data.principals, data.sourceAddress)
+	t.Logf("Testing SSH certificate with config:\n %s", config)
+	r.Test(t, r.TestCase{
+		Providers: testProviders,
+		Steps: []r.TestStep{
+			r.TestStep{
+				Config: config,
+				Check: func(s *terraform.State) error {
+					err := checkSshCertificate(t, &data, s)
+					if err != nil {
+						return err
+					}
+					return nil
+				},
+				ExpectNonEmptyPlan: true,
+			},
+		},
+	})
+}
+
+func TestSshCertLocalPublicKey(t *testing.T) {
+	t.Log("Testing creating ssh certificate")
+
+	data := getTestData()
+	data.publicKeyMethod = "local"
+
+	// data.principals only holds the values for principals, actually we are testing "principal" attribute defined at the resource.
+	config := fmt.Sprintf(tppSshCertResourceTest, tokenSshCertProv, data.keyId, data.template, data.publicKeyMethod, data.validityPeriod, data.principals, data.sourceAddress)
```

Comment:
Could we use the new resources that works with principals on this test?

Reply:
We are covering that use code in the test ` TestSshCertLocalPublicKeyNewAttrPrincipals`